### PR TITLE
Feature/ktor for url parsing

### PR DESCRIPTION
**Pull request by fedyiv**

### CODE REVIEW

File: src/jvmTest/kotlin/com/dxc/ssi/agent/transport/EndpointTest.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package com.dxc.ssi.agent.transport
```
Comment (kkamyczek) [inline src/jvmTest/kotlin/com/dxc/ssi/agent/transport/EndpointTest.kt:1]: Why those tests are commented out?

File: src/commonMain/kotlin/com/dxc/ssi/agent/transport/WebSocketTransportImpl.kt [hunk 2]
```diff
@@ -31,20 +31,12 @@ class WebSocketTransportImpl : Transport {
...
-        val host = parseHostFromEndpoint(connection.endpoint)
-        val port = parsePortFromEndpoint(connection.endpoint)
-        val path = parsePathFromEndpoint(connection.endpoint)
-        val protocol = parseProtocolFromEndpoint(connection.endpoint)
+        println("Before sending message to endpoint: ${connection.endpoint}")
 
-        println("host = $host, port = $port, path = $path, protocol = $protocol")
-
-        if (protocol != "ws")
+        if (!(connection.endpoint.protocol == URLProtocol.WS || connection.endpoint.protocol == URLProtocol.WSS))
```
Comment (kkamyczek) [inline src/commonMain/kotlin/com/dxc/ssi/agent/transport/WebSocketTransportImpl.kt:36]: It's not about this line. Are we planning to have tests with WSS?

**Review by kkamyczek — COMMENTED** [review]
Done

File: src/commonMain/kotlin/com/dxc/ssi/agent/transport/WebSocketTransportImpl.kt (hunk 2 above)
Comment (fedyiv) [reply src/commonMain/kotlin/com/dxc/ssi/agent/transport/WebSocketTransportImpl.kt:36]: Yes, we have separate task for WSS support

File: src/jvmTest/kotlin/com/dxc/ssi/agent/transport/EndpointTest.kt (hunk 1 above)
Comment (fedyiv) [reply src/jvmTest/kotlin/com/dxc/ssi/agent/transport/EndpointTest.kt:1]: Because they had only temporal meaning, and not needed here. I will remove them.

**Review by kkamyczek — APPROVED** [review]

**Pull request merged**